awx/main/models/workflow.py

```diff
@@ -686,17 +698,59 @@ def _get_parent_field_name(self):
     def approve(self, request=None):
         self.status = 'successful'
         self.save()
+        self.send_approval_notification('approved')
         self.websocket_emit_status(self.status)
         schedule_task_manager()
         return reverse('api:workflow_approval_approve', kwargs={'pk': self.pk}, request=request)
 
     def deny(self, request=None):
         self.status = 'failed'
         self.save()
+        self.send_approval_notification('denied')
         self.websocket_emit_status(self.status)
         schedule_task_manager()
         return reverse('api:workflow_approval_deny', kwargs={'pk': self.pk}, request=request)
 
+    def signal_start(self, **kwargs):
+        can_start = super(WorkflowApproval, self).signal_start(**kwargs)
+        self.send_approval_notification('running')
+        return can_start
+
+    def send_approval_notification(self, status):
+        from awx.main.tasks import send_notifications  # avoid circular import
+        if self.workflow_job_template is None:
+            return
+        for nt in self.workflow_job_template.notification_templates["approvals"]:
+            try:
+                (notification_subject, notification_body) = self.build_notification_message(nt, status)
+            except AttributeError:
+                raise NotImplementedError("build_notification_message() does not exist" % status)
+
+            def send_it(local_nt=nt, local_subject=notification_subject, local_body=notification_body):
+                def _func():
+                    send_notifications.delay([local_nt.generate_notification(local_subject, local_body).id],
+                                             job_id=self.id)
+                return _func
+            connection.on_commit(send_it())
+
+    def build_notification_message(self, nt, status):
+        subject = []
+        workflow_url = urljoin(settings.TOWER_URL_BASE, '/#/workflows/{}'.format(self.workflow_job.id))
+        subject.append(('The approval node "{}"').format(self.workflow_approval_template.name))
+        if status == 'running':
+            subject.append((' is running. This node can be approved or denied at: {}').format(workflow_url))
+        if status == 'approved':
+            subject.append((' was approved. {}').format(workflow_url))
+        if status == 'timed_out':
+            subject.append((' has timed out. {}').format(workflow_url))
+        elif status == 'denied':
+            subject.append((' was denied. {}').format(workflow_url))
+        subject = " ".join(subject)
+        body = self.notification_data()
+        body['body'] = subject
```

Comment:
This looks great 👍 @jladdjr you have any other thoughts here?

Reply:
@ryanpetrello - @beeankha and I took a look at this last week - this looks like it's doing the right thing 👍